src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -506,6 +506,42 @@ private TypeSpec CreateEnumerableSpec(TypeParseInfo typeParseInfo)
                 };
             }
 
+            private bool ConvertTo(ITypeSymbol source, ITypeSymbol dest)
+            {
+                var conversion = _typeSymbols.Compilation.ClassifyConversion(source, dest);
+                return conversion.IsReference && conversion.IsImplicit;
+            }
+
+            private bool IsUnsupportedType(ITypeSymbol type)
+            {
+                if (SymbolEqualityComparer.Default.Equals(_typeSymbols.IntPtr, type)  ||
+                    SymbolEqualityComparer.Default.Equals(_typeSymbols.UIntPtr, type) ||
+                    SymbolEqualityComparer.Default.Equals(_typeSymbols.SerializationInfo, type) ||
+                    ConvertTo(type, _typeSymbols.MemberInfo) ||
```

Comment:
```suggestion
                    CanConvertTo(type, _typeSymbols.MemberInfo) ||
```

Reply:
Could we extend this for all types in System.Reflection somehow? For example, `ParameterInfo` is not a subtype of `MemberInfo`.

Reply:
I am seeing `ParameterInfo` is the only missing one. I cann this. any idea why text json didn't do it as we were trying to do what text json doing?

Reply:
It's an omission, it should be added there as well eventually (though probably `ParameterInfo` is not a super common scenario compared to `Type` and `MemberInfo`).

Reply:
this is renamed to `IsAssignableTo`. Also I added a support for `ParameterInfo` too.